watertap/property_models/NDMA_prop_pack.py

```diff
@@ -0,0 +1,781 @@
+                    f"check the output logs for more information."
+                )
+
+        # ---------------------------------------------------------------------
+        # If input block, return flags, else release state
+        if state_vars_fixed is False:
+            if hold_state is True:
+                return flags
+            else:
+                self.release_state(flags)
+
+    def release_state(self, flags, outlvl=idaeslog.NOTSET):
+        """
+        Method to release state variables fixed during initialisation.
+
+        Keyword Arguments:
+            flags : dict containing information of which state variables
+                    were fixed during initialization, and should now be
+                    unfixed. This dict is returned by initialize if
+                    hold_state=True.
+            outlvl : sets output level of of logging
+        """
+        # Unfix state variables
+        init_log = idaeslog.getInitLogger(self.name, outlvl, tag="properties")
+        revert_state_vars(self, flags)
+        init_log.info_high("{} State Released.".format(self.name))
+
+    def calculate_state(
+        self,
+        var_args=None,
+        hold_state=False,
+        outlvl=idaeslog.NOTSET,
+        solver=None,
+        optarg=None,
+    ):
+        """
+        Solves state blocks given a set of variables and their values. These variables can
+        be state variables or properties. This method is typically used before
+        initialization to solve for state variables because non-state variables (i.e. properties)
+        cannot be fixed in initialization routines.
+
+        Keyword Arguments:
+            var_args : dictionary with variables and their values, they can be state variables or properties
+                       {(VAR_NAME, INDEX): VALUE}
+            hold_state : flag indicating whether all of the state variables should be fixed after calculate state.
+                         True - State variables will be fixed.
+                         False - State variables will remain unfixed, unless already fixed.
+            outlvl : idaes logger object that sets output level of solve call (default=idaeslog.NOTSET)
+            solver : solver name string if None is provided the default solver
+                     for IDAES will be used (default = None)
+            optarg : solver options dictionary object (default={})
+
+        Returns:
+            results object from state block solve
+        """
+        # Get logger
+        solve_log = idaeslog.getSolveLogger(self.name, level=outlvl, tag="properties")
+
+        # Initialize at current state values (not user provided)
+        self.initialize(solver=solver, optarg=optarg, outlvl=outlvl)
+
+        # Set solver and options
+        opt = get_solver(solver, optarg)
+
+        # Fix variables and check degrees of freedom
+        flags = (
+            {}
+        )  # dictionary noting which variables were fixed and their previous state
```

Comment:
```suggestion
         # dictionary noting which variables were fixed and their previous state
        flags = {}
```
Putting comments at the end of the line is not very black friendly